### PR TITLE
feature/74 | Added EC2 Security Group RFC 1918 plugin, spec file

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'EC2 Security Group RFC 1918',
+    category: 'EC2',
+    description: 'Ensures EC2 security groups are not configured to allow inbound traffic from RFC-1918 CIDRs',
+    more_info: 'RFC-1918 IP addresses are considered reserved private addresses and should not be used in security groups.',
+    link: 'https://www.cloudconformity.com/knowledge-base/aws/EC2/security-group-rfc1918.html',
```
Comment (giorod3) [inline plugins/aws/ec2/securityGroupRfc1918.js:9]: Use AWS links only please.

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    recommended_action: 'Modify the security group to ensure the private reserved addresses are not allowed for inbound traffic',
+    apis: ['EC2:describeSecurityGroups'],
+    settings: {
+        privateCidrs: {
+            name: 'EC2 RFC 1918 CIDR Addresses',
+            description: 'A comma-delimited list of CIDRs that indicates reserved private addresses',
+            regex: '[a-zA-Z0-9,]',
+            default: ['10.0.0.0/8', '172.16.0.0/12', '192.168.0.0/16']
```
Comment (giorod3) [inline plugins/aws/ec2/securityGroupRfc1918.js:17]: change this to be a string

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                        if (cidrIp && privateCidrs.includes(cidrIp)) {
+                            if(!privateCidrsFound.includes(cidrIp)) {
+                                privateCidrsFound.push(cidrIp);
+                            }
+                        }
+                    }
+
+                    if(privateCidrsFound.length) {
+                        helpers.addResult(results, 2,
+                            'Security group :' + group.GroupName + ': allows inbound access for these reserved private address: ' + privateCidrsFound.join(', '),
```
Comment (giorod3) [inline plugins/aws/ec2/securityGroupRfc1918.js:69]:
```suggestion
                            'Security group :' + group.GroupName + ': allows inbound access for these reserved private addresses: ' + privateCidrsFound.join(', '), 
```

**Review by giorod3 — CHANGES_REQUESTED** [review]

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+var async = require('async');
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'EC2 Security Group RFC 1918',
```
Comment (giorod3) [inline plugins/aws/ec2/securityGroupRfc1918.js:5]:
```suggestion
    title: 'Open RFC 1918',
```

**Review by giorod3 — APPROVED** [review]

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            }
+
+            for (var g in describeSecurityGroups.data) {
+                var group = describeSecurityGroups.data[g];
+                var resource = 'arn:' + awsOrGov + ':ec2:' + region + ':' + group.OwnerId + ':security-group/' + group.GroupId;
+                var privateCidrsFound = [];
+
+                if (!group.IpPermissions || !group.IpPermissions.length) {
+                    helpers.addResult(results, 0,
+                        'Security group :' + group.GroupName + ': does not have any ip permission', region, resource);
```
Comment (matthewdfuller) [inline plugins/aws/ec2/securityGroupRfc1918.js:54]:
```suggestion
                        'Security group :' + group.GroupName + ': does not have any IP permissions', region, resource);
```

File: plugins/aws/ec2/securityGroupRfc1918.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                        if (cidrIp && privateCidrs.includes(cidrIp)) {
+                            if(!privateCidrsFound.includes(cidrIp)) {
+                                privateCidrsFound.push(cidrIp);
+                            }
+                        }
+                    }
+
+                    if(!privateCidrsFound.length) {
+                        helpers.addResult(results, 0,
+                            'Security group "' + group.GroupName + '" is configured to deny any reserved private address',
```
Comment (matthewdfuller) [inline plugins/aws/ec2/securityGroupRfc1918.js:72]:
```suggestion
                            'Security group "' + group.GroupName + '" is not configured to allow traffic from any reserved private addresses',
```

**Pull request merged**